### PR TITLE
2rp program prp and run helloworld

### PR DESCRIPTION
**Pull request by houlz0507**
Test: http://confluence.xilinx.com/display/~lizhih/EXAMPLES

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c [hunk 1]
```diff
@@ -732,7 +732,7 @@ int xocl_init_mem(struct xocl_drm *drm_p)
 
 		drm_mm_init(drm_p->mm[i], mem_data->m_base_address,
 				ddr_bank_size - reserved1 - reserved2);
-		drm_p->mm_p2p_off[i] = ddr_bank_size * i;
+		drm_p->mm_p2p_off[i + 1] = drm_p->mm_p2p_off[i] + ddr_bank_size;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c]: some ddr banks are skipped, not sure how would it work here?

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_sysfs.c [hunk 2]
```diff
@@ -185,14 +185,23 @@ static ssize_t p2p_enable_store(struct device *dev,
...
+	else {
+		size = xocl_get_ddr_channel_size(xdev) *
+			xocl_get_ddr_channel_count(xdev); /* GB */
+	}
+
 	size = (ffs(size) == fls(size)) ? (fls(size) - 1) : fls(size);
 	size = enable ? (size + 10) : (XOCL_PA_SECTION_SHIFT - 20);
 	if (xocl_pci_rebar_size_to_bytes(size) == curr_size) {
-		xocl_info(&pdev->dev, "p2p is enabled, bar size %d M",
+		if (enable)
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_sysfs.c]: Need '{' and '}' for multi-line if-else statement.

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp [hunk 3]
```diff
@@ -108,6 +108,41 @@ static int scanDevices(bool verbose, bool json)
     return 0;
 }
 
+static bool match_id(DSAInfo& dsa, std::string& id)
+{
+printf("DSA %s id %s, ts %lx\n", dsa.uuid.c_str(), id.c_str(), dsa.timestamp);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp]: remove this line?

Comment (houlz0507) [reply src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp]: removed

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_sysfs.c (hunk 2 above)
Comment (houlz0507) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_sysfs.c]: added

**Pull request merged**